### PR TITLE
buildchain: provide more error output in container run/build crashes

### LINKED ISSUE #1263: Provide additional output to docker build/run in buildchain
**Component**: buildchain

**Why this is needed**:

Currently we don't provide any significant output when the docker API fails, but we could using the exceptions we catch.

**What should be done**:

More debug output on docker API exceptions.

**Implementation proposal** (strongly recommended):

**Test plan**:

Add `RUN my_first_pony` to, for example, `images/keepalived/Dockerfile`.

* Before: one line output, with only error code and no significant output
* After: Whole `docker build`ish log.

### PR DESCRIPTION
**Pull request by jbertran**
**Component**:

buildchain

**Context**: 

More verbose docker errors.

**Summary**:

* More verbose `ContainerError` (`docker run`)
* More verbose `BuildError` (`docker build`)

**Acceptance criteria**: 

More verbose errors.

---

Closes: #1263

### CODE REVIEW

**Comment by jbertran** [comment]
@bert-e reset

**Review by gdemonet — DISMISSED** [review]
LGTM - may want to refactor this when we move to `buildah`

**Review by gdemonet — DISMISSED** [review]
Perfect!

**Comment by jbertran** [comment]
@bert-e reset

File: buildchain/buildchain/docker_command.py [hunk 1]
```diff
@@ -71,7 +72,21 @@ def decorated_task(*args: Any, **kwargs: Any) -> Optional[TaskError]:
...
-                return TaskError(err)
+                # Output container build log on BuildError
+                if isinstance(err, BuildError):
+                    output_lines = [
+                        item['stream']
+                        if 'stream' in item else item['error']
+                        for item in err.build_log
+                    ]
+                    log = ''.join(output_lines)
+                    err_msg = '{}. Build log:\n{}'.format(str(err), log)
```
Comment (slaperche-scality) [inline buildchain/buildchain/docker_command.py]: Why not `err_msg = '{}:\n{}'.format(str(err), log)` to be consistent with `ContainerError`?
Or use sth like `err_msg = '{}. Exec error:\n{}'.format(str(err), err.stderr)` for `ContainerError`

File: buildchain/buildchain/docker_command.py [hunk 2]
```diff
@@ -56,7 +56,31 @@ def bind_ro_mount(source: Path, target: Path) -> Mount:
...
+    return '{}. Build log:\n{}'.format(str(build_error), log)
+
+
+def container_error_handler(container_error: ContainerError) -> str:
+    """String formatting exception handler for Docker API ContainerError."""
+    return '{}:\n{}'.format(str(container_error), container_error.stderr)
+
+
+def task_error(
+        expected_exn: Type[Exception],
```
Comment (slaperche-scality) [inline buildchain/buildchain/docker_command.py]: minor: continuation line are usually indented by four spaces in this file.

**Review by slaperche-scality — CHANGES_REQUESTED** [review]
Your last commit really make the code more legible :)

**Comment by jbertran** [comment]
@bert-e reset

**Review by slaperche-scality — APPROVED** [review]

**Comment by jbertran** [comment]
@bert-e reset

**Comment by jbertran** [comment]
@bert-e approve

**Pull request merged**